### PR TITLE
diff/print-only: Report file names

### PR DESCRIPTION
**Pull request by abhinav**
Include names of matched files as we print comments to stderr.
This turns an unidentifiable stream of messages like,

    Replace foo with bar
    Replace foo with bar
    Replace baz with qux

Into more standard linter-style messages.

    foo.go:Replace foo with bar
    bar.go:Replace foo with bar
    foo.go:Replace baz with qux

In doing this, I went over the different sample patches we have in the
repository and came up with some best practices around this.
Seeing as these messages are meant to be shown to the user,
these should aim to be short and to the point.
So I suggest,

- Keep them on a single line
- Don't use "this patch" in the message
- Use imperative mood ("replace X with Y")

Basically, provide just enough information for a user to know what
happened.

### CODE REVIEW

**Review by sywhang — APPROVED** [review]

**Comment by abhinav** [comment]
My bad on merging this. I saw the green for approved, and did not notice that the build was red. Fixing in #75.
